platform/files.go

```diff
@@ -87,6 +87,11 @@ type PreviousImageRunImageMetadata struct {
 	Reference string `json:"reference" toml:"reference"`
 }
 
+type RunImage struct {
+	Reference string                   `toml:"reference"`
+	Target    buildpack.TargetMetadata `json:"target" toml:"target"`
```

Comment:
`buildpack.TargetMetadata` is subtly different from `RunImage.Target` in that there can be multiple distributions:

```
type TargetMetadata struct {
	Os            string                 `json:"os" toml:"os"`
	Arch          string                 `json:"arch" toml:"arch"`
	ArchVariant   string                 `json:"arch-variant" toml:"arch-variant"`
	Distributions []DistributionMetadata `json:"distributions" toml:"distributions"` <- a single image can only have one of these
}
```

We might want to make a `platform.TargetMetadata` that looks like:

```
type TargetMetadata struct {
	Os            string                 `json:"os" toml:"os"`
	Arch          string                 `json:"arch" toml:"arch"`
	ArchVariant   string                 `json:"arch-variant" toml:"arch-variant"`
	Distribution  buildpack.DistributionMetadata `json:"distribution" toml:"distribution"`
}
```

I think this could make `TargetMetadata.Satisfies` simpler as we'll only have to check that "buildpack.Distributions" contains "platform.Distribution" instead of list equality.

Reply:
thanks. i think you're right, and that let me to update a few other things,  so it's now `IsSatisfiedBy` b/c the method needs to know about both the TargetMetaData types (in platform and in buildpack) so it needs to be in the 'higher' package up the dependency graph...